Automattic/wp-calypso
# Add the Premium plan upsell to the domain/plan upsell modals.

Fix Automattic/growth-foundations#286

## Proposed Changes

This PR is the final closure for a series of PRs addressing Automattic/growth-foundations#286, adding the Premium plan upsell to the domain/plan upsell modals. For the updated design, please refer to:

* The experiment post: pbxNRc-3pt-p2, even though it's no longer an experiment based on the request.
* The project thread: peP6yB-1II-p2#comment-856

Free domain/Free plan:
<img width="643" alt="CleanShot 2023-12-26 at 13 46 44@2x" src="https://github.com/Automattic/wp-calypso/assets/1842898/81d924d1-b532-4154-acf7-895048140f14">

Paid domain/Free plan:
<img width="673" alt="CleanShot 2023-12-26 at 13 47 07@2x" src="https://github.com/Automattic/wp-calypso/assets/1842898/fcc4735a-4923-4a76-88a1-52532381e06b">

A paid plan is requried:
<img width="689" alt="CleanShot 2023-12-26 at 13 47 29@2x" src="https://github.com/Automattic/wp-calypso/assets/1842898/46c272e9-0b6c-459a-a096-67781dd2b32e">

I think the responsiveness still has room to improve, but it's not in the scope of this iteration.

## Testing Instructions



* In `/start/onboarding-pm`:
    * Pick a free domain and the Free plan; confirm that the Free domain/Free plan modal works as expected and matches the design.
    * Pick a paid domain and the Free plan; confirm that the Paid domain/Free plan modal works as expeced and matches the design.
* In `/start/`:
    * Pick a paid domain and the Free plan; confirm that the paid-plan-is-requred modal works as expected and matches the design.

## Pre-merge Checklist
